core/core-impl/src/test/java/ai/timefold/solver/core/impl/score/director/AbstractScoreDirectorSemanticsTest.java

```diff
@@ -141,6 +141,8 @@ void constraintPresentEvenIfNoMatches() {
             scoreDirector.beforeVariableChanged(entity, "value");
             entity.setValue(null);
             scoreDirector.afterVariableChanged(entity, "value");
+            // FULL_ASSERT add variable listeners for tracking
+            scoreDirector.triggerVariableListeners();
             var score2 = scoreDirector.calculateScore();
```

Comment:
This concerns me.
Are you saying we need to do something *new* in order to use this functionality? 
Why would this code need it, if it doesn't run any of the assert modes?

Reply:
This code was introduced before `FULL_ASSERT` and `TRACKED_FULL_ASSERT` were seperated. `TRACKED_FULL_ASSERT` adds it own variable listeners, so it need the variable listeners to be trigerred to not throw an exception about missed before... and after... calls.

Reply:
Which effectively means it will *break* everything which didn't already do this.
`ScoreDirector` is public, its behavior is set in stone.
Therefore this is not acceptable.